DRODLib/CurrentGame.cpp

```diff
@@ -675,6 +675,24 @@ int CCurrentGame::EvalPrimitive(ScriptVars::PrimitiveType ePrimitive, const vect
 			const SQUARE& square = this->pRoom->pPathMap[movement]->GetSquare(x, y);
 			return square.dwTargetDist;
 		}
+		case ScriptVars::P_BrainDist:
+		{
+			int tX = params[0];
+			int tY = params[1];
+			int x = params[2];
+			int y = params[3];
+			MovementType movement = (MovementType)params[4];
+
+			if (!this->pRoom->IsValidColRow(tX, tY) || !this->pRoom->IsValidColRow(x, y) ||
+				movement >= NumMovementTypes) {
+				return -1;
+			}
+
+			//Make sure pathmap exists
+			this->pRoom->CreatePathMap(tX, tY, movement);
```

Comment:
Two questions:

Is the intent to override the brained target? If not, wouldn't this approach collide with operations around pre-existing or other pathmaps in the room that manage the same movement type? If they're not intended to collide, wouldn't we need to restore any pre-existing pathmap after performing this calculation?

Also, this approach sounds expensive if it's used more than once per turn!

Reply:
Did some more testing it this does weird stuff if it's called before monsters making brained moves. So a rethink is definitely in order. We might paths for this held somewhere else, which can also allow caching to reduce the cost of doing this multiple times.

Reply:
I think this problem technically also exists with `Nearest open edge to me`, but since edge-finding uses the `FORCE` movement types, it doesn't affect monster pathing. Then when a horn is used that sets the target so it doesn't cause a problem there.

Reply:
Thanks for checking on this. How about we use dedicated movement slots in the room data object to store these ad-hoc scripted pathmaps, say, "EPHEMERAL + x" movement index, so they won't clobber the base types.

Reply:
Tried a few possible solutions for this. Ultimately, the smoothest one was to add a new set of pathmaps in `CDbRoom` that I've called `pExtraPathMap`. For the moment it's only used by the `_BrainDist` logic, but could be used elsewhere if needed.